elephant/signal.py

```
@@ -0,0 +1,147 @@
+        3. compute the periodogram of each segment
+        4. average the obtained periodograms to yield PSD estimate
+    These steps are implemented in `scipy.signal`, and this function is a
+    wrapper which provides a proper set of parameters to
+    `scipy.signal.welch()`. Some parameters for scipy.signal.welch(), such as
+    `nfft`, `detrend`, `window`, `return_onesided` and `scaling`, also works
+    for this function.
+
+    Parameters
+    ----------
+    signal: Neo AnalogSignalArray or Quantity array or Numpy ndarray
+        Time series data, of which PSD is estimated. When a Quantity array or
+        Numpy ndarray is given, sampling frequency should be given through the
+        keyword argument `fs`, otherwise the default value (`fs=1.0`) is used.
+    num_seg: int, optional
+        Number of segments. The length of segments is adjusted so that
+        overlapping segments cover the entire stretch of the given data. This
+        parameter is ignored if *len_seg* or *freq_res* is given. Default is 8.
+    len_seg: int, optional
+        Length of segments. This parameter is ignored if *freq_res* is given.
+        Default is None (determined from other parameters).
+    freq_res: Quantity or float, optional
+        Desired frequency resolution of the obtained PSD estimate in terms of
+        the interval between adjacent frequency bins. When given as a float, it
+        is taken as frequency in Hz. Default is None (determined from other
+        parameters).
+    overlap: float, optional
+        Overlap between segments represented as a float number between 0 (no
+        overlap) and 1 (complete overlap). Default is 0.5 (half-overlapped).
+    fs: Quantity array or float, optional
+        Specifies the sampling frequency of the input time series. When the
+        input is given as an AnalogSignalArray, the sampling frequency is taken
+        from its attribute and this parameter is ignored. Default is 1.0.
```

This is something to fix later, but it is dangerous to ignore arguments. If `fs` is specified explicitly and is different from the sampling frequency of the signal, the code should raise an Exception.